feat!: update configuration

gentlementlegen opened the pull request:
Resolves #8

<!--
- You must link the issue number e.g. "Resolves #1234"
- Please do not replace the keyword "Resolves" https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

gentlementlegen committed:
feat!: updated the config to match the new version

gentlementlegen committed:
chore: added data purge config

gentlementlegen committed:
chore: added mock for valid configuration

gentlementlegen committed:
chore: moved deps to external

gentlementlegen committed:
v2.0.0

gentlementlegen commented:
Main changes:
- updated the configuration to match the V2 new configuration for the bot
- moved dependencies to peer dependencies since this library is meant to be included inside others and not on its own (saved 200kb as well in the bundle)
- updated the tests and package version accordingly

gentlementlegen committed:
chore: updated default configuration

gentlementlegen committed:
chore: updated action for publishing

gentlementlegen committed:
chore: exporting base incentive configuration

gentlementlegen committed:
v2.0.1

gentlementlegen committed:
v2.0.2

0x4007 commented on .github/ubiquibot-config.yml:
```diff
@@ -7,3 +7,70 @@ command-settings:
+      td: 1
+      hr: 0
+    multipliers:
+      - type: [ ISSUE, ISSUER, SPECIFICATION ]
```
I'm still confused about this syntax. Can you explain what these are meant to express?

gentlementlegen replied:
Dynamic configurable replacement of
https://github.com/ubiquibot/comment-incentives/blob/525fcdccc8f0d9032a41cdaed3b4fbc6514d44b6/src/handlers/issue/comment-scoring-by-contribution-style.ts#L7

gitcoindev replied:
If I understand correctly, the syntax is: first value ISSUE / REVIEW , depends on location where the comment is located. Second value on the person: ISSUER /  ASSIGNEE / COLLABORATOR / CONTRIBUTOR . Third value depends on the action COMMENTED / SPECIFICATION / TASK (perhaps SPECIFIED / TASK_CREATED would be more clear but I think we can leave as is) . Therefore in general this is a triple [ location / person / action ] . Would be good to put this information into a readme or instruction somewhere.

gentlementlegen replied:
@gitcoindev you are correct, for now it is that way. But it is made so that we can extend it with more values if we need to. Any combination not found within the list defaults to a 0 multiplier to avoid accidents.

0x4007 commented on src/types/configuration/user-extractor-config.ts:
```diff
@@ -0,0 +1,8 @@
+
+export const userExtractorConfigurationType = Type.Object({
+  enabled: Type.Boolean({ default: true }),
+  redeemTask: Type.Boolean({ default: true }),
```
What is redeem task?

gentlementlegen replied:
If you disable this, users won't be able to collect the reward associated to the issue

gitcoindev commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@ubiquibot/configuration",
-  "version": "1.1.0",
+  "version": "2.0.1",
```
Why not 2.0.0 ?

gentlementlegen replied:
To make sure the package works properly I published `2.0.0` so to avoid conflicts with `release-please` I bumped it one patch version up.

gentlementlegen replied:
And afterwards had to fix a small bug, so here we are in `2.0.2`
https://www.npmjs.com/package/@ubiquibot/configuration

gitcoindev commented on src/types/configuration/github-comment-config.ts:
```diff
@@ -0,0 +1,9 @@
+export const githubCommentConfigurationType = Type.Object({
+  enabled: Type.Boolean({ default: true }),
+  post: Type.Boolean({ default: true }),
+  debug: Type.Boolean({ default: false }),
```
It would be good also to document what each of those configuration types mean, can be above the code.

gentlementlegen replied:
That's a good idea, I will add comments.

gitcoindev reviewed (changes requested):
I left a few comments to add documentation but in general a good job!

gitcoindev commented:
@gentlementlegen I noticed Jests tests are failing, please look into this as well https://github.com/ubiquibot/configuration/actions/runs/8826720734/job/24232850716?pr=9

gentlementlegen committed:
chore: added comments for configurations

gentlementlegen committed:
chore: fixed tests

gentlementlegen committed:
chore: renamed configuration

gentlementlegen committed:
v2.0.3

gentlementlegen commented:
> @gentlementlegen I noticed Jests tests are failing, please look into this as well https://github.com/ubiquibot/configuration/actions/runs/8826720734/job/24232850716?pr=9

@gitcoindev Tests are fixed and I should have addressed all your comments
https://github.com/ubiquibot/configuration/actions/runs/8845050329/job/24288169884

gentlementlegen committed:
feat!: added plugins type

gentlementlegen committed:
chore: fixed cspell ignores
